baremetal/src/Runtime.cc

```diff
@@ -55,16 +59,48 @@ class Connection : public ebbrt::TcpHandler {
 }  // namespace
 
 void ebbrt::runtime::Init() {
-  auto reader = boot_fdt::Get();
-  auto offset = reader.GetNodeOffset("/runtime");
-  auto ip = reader.GetProperty32(offset, "address");
-  auto port = reader.GetProperty16(offset, "port");
-  auto allocation_id = reader.GetProperty16(offset, "allocation_id");
+  auto cmdline = std::string(ebbrt::multiboot::CmdLine());
+  auto host = std::string("host_addr=");
```

Comment:
when you say `"bar"``, it places a static string in the binary. When you store that to a `std::string` it might do a heap allocation to store that string. You pretty quickly thereafter stop using these variables so these heap allocations are relatively costly.  You should be able to avoid storing them to strings.